src/app/[locale]/(main)/profile/[id]/profile/actions/fetch-user-profile.tsx

```diff
@@ -0,0 +1,36 @@
+import { getToken } from "@/lib/auth"
```

Comment:
all action server must be mark as "use server" and function have "action" as suffix (ex: `getUserProfileAction()`)